src/catch2/generators/catch_generators.hpp

```diff
@@ -219,6 +220,7 @@ namespace Detail {
         }
 
         auto const& generator = static_cast<IGenerator<UnderlyingType> const&>( *tracker.getGenerator() );
+        getResultCapture().trackGeneratorState(GeneratorInfo(generatorName, argumentsDescription, lineInfo, generator.currentElementAsString()));
```

Comment:
I was wondering if there's maybe a "cheaper" or better way to only get the generator states, if the test fails.